src/applications/mhv-medical-records/containers/App.jsx

```diff
@@ -158,53 +159,60 @@ const App = ({ children }) => {
   }
 
   return (
-    <RequiredLoginView
-      user={user}
-      serviceRequired={[backendServices.MEDICAL_RECORDS]}
-    >
-      {isMissingRequiredService(user.login.currentlyLoggedIn, userServices) || (
-        <>
-          {phase0p5Flag && <MhvSecondaryNav />}
-          <div
-            ref={measuredRef}
-            className="vads-l-grid-container vads-u-padding-left--2"
-          >
-            {mhvMrDown === externalServiceStatus.down ? (
-              <>
-                {atLandingPage && <MrBreadcrumbs />}
-                <h1 className={atLandingPage ? null : 'vads-u-margin-top--5'}>
-                  Medical records
-                </h1>
-                <DowntimeNotification
-                  appTitle={downtimeNotificationParams.appTitle}
-                  dependencies={[
-                    externalServices.mhvMr,
-                    externalServices.mhvPlatform,
-                    externalServices.global,
-                  ]}
-                  render={renderMHVDowntime}
-                />
-              </>
-            ) : (
-              <HeaderSectionProvider>
-                <MrBreadcrumbs />
-                <div className="vads-l-row">
-                  <div className="medium-screen:vads-l-col--8">{children}</div>
-                </div>
-              </HeaderSectionProvider>
-            )}
-            <va-back-to-top
-              class="no-print"
-              hidden={isHidden}
-              data-dd-privacy="mask"
-              data-dd-action-name="Back to top"
-              data-testid="mr-back-to-top"
-            />
-            <ScrollToTop />
-            <PhrRefresh statusPollBeginDate={statusPollBeginDate} />
-          </div>
-        </>
-      )}
+    <RequiredLoginView user={user}>
+      <MhvServiceRequiredGuard
+        user={user}
+        serviceRequired={[backendServices.MEDICAL_RECORDS]}
+      >
+        {isMissingRequiredService(
+          user.login.currentlyLoggedIn,
+          userServices,
+        ) || (
```

Comment:
We should no longer need the `isMissingRequiredService` function.

**Additionally**, please get rid of the `useEffect` immediately above that function. That was pre-Phase 0 code, and we no longer need it.